File: boosting/src/utils/RandomUtils.java
package utils;

import java.util.Random;

public class RandomUtils {

    private final static RandomUtils instance = new RandomUtils();

    private final Random random;

    private RandomUtils() {
        random = new Random();
    }

    public static RandomUtils getInstance() {
        return instance;
    }

    public int nextInt() {
        return random.nextInt();
    }

    public int nextInt(int bound) {
        return random.nextInt(bound);
    }

    public double nextDouble() {
        return random.nextDouble();
    }
}

File: boosting/src/utils/RandomUtils.kt
package utils

import java.util.Random

class RandomUtils private constructor() {

    private val random: Random

    init {
        random = Random()
    }

    fun nextInt(): Int {
        return random.nextInt()
    }

    fun nextInt(bound: Int): Int {
        return random.nextInt(bound)
    }

    fun nextDouble(): Double {
        return random.nextDouble()
    }

    fun nextDouble(maxRange: Double): Double {
        return nextDouble() * maxRange
    }

    companion object {

        val instance = RandomUtils()
    }
}
